On macos 10.15:

```
=== RUN   TestInputDir_RespectsUlimit
    inputdir_unix_test.go:49: 
                Error Trace:    integration_test.go:57
                                                        inputdir_unix_test.go:49
                Test:           TestInputDir_RespectsUlimit
    inputdir_unix_test.go:49: 
                Error Trace:    integration_test.go:58
                                                        inputdir_unix_test.go:49
                Error:          Not equal: 
                                expected: ""
                                actual  : "{\"level\":\"error\",\"error\":\"failed to render template /private/tmp/nix-build-gomplate-3.11.2.drv-0/ulimittestfiles-1626002131/in/file_3: failed to open output file '/private/tmp/nix-build-gomplate-3.11.2.drv-0/ulimittestfiles-1626002131/out/file_3' for writing: open /private/tmp/nix-build-gomplate-3.11.2.drv-0/ulimittestfiles-1626002131/out/file_3: too many open files\",\"time\":\"2022-08-09T20:43:41Z\"}\n"
                                
                                Diff:
                                --- Expected
                                +++ Actual
                                @@ -1 +1,2 @@
                                +{"level":"error","error":"failed to render template /private/tmp/nix-build-gomplate-3.11.2.drv-0/ulimittestfiles-1626002131/in/file_3: failed to open output file '/private/tmp/nix-build-gomplate-3.11.2.drv-0/ulimittestfiles-1626002131/out/file_3' for writing: open /private/tmp/nix-build-gomplate-3.11.2.drv-0/ulimittestfiles-1626002131/out/file_3: too many open files","time":"2022-08-09T20:43:41Z"}
                                 
                Test:           TestInputDir_RespectsUlimit
    inputdir_unix_test.go:53: assertion failed: 32 (numfiles int) != 23 (int)
--- FAIL: TestInputDir_RespectsUlimit (0.02s)
```

Could do a

```
  preCheck = ''
    ulimit -n 1024
  '';
```

as done in some other places

--- a/pkgs/development/tools/gomplate/default.nix
+++ b/pkgs/development/tools/gomplate/default.nix
@@ -2,17 +2,17 @@
 
 buildGoModule rec {
   pname = "gomplate";
-  version = "3.10.0";
+  version = "3.11.2";
   owner = "hairyhenderson";
   rev = "v${version}";
 
   src = fetchFromGitHub {
     inherit owner rev;
     repo = pname;
-    sha256 = "0dbi9saxbwcvypxc0s656ln9zq2vysx8dhrcz488nmy6rcpqiiah";
+    sha256 = "sha256-NIepoz1JToaX2EJCL/kqkpBJigJVy2Tkz0jGn4ukfvI=";
   };
 
-  vendorSha256 = "0rvki8ghlbbaqgnjfsbs1jswj08jfzmnz9ilynv2c6kfkx9zs108";
+  vendorSha256 = "sha256-fXbwNX+GoujciZVxxe7Tl21MxWhyAD4cW/p8PCAAElw=";
 
   postPatch = ''
     # some tests require network access
@@ -22,6 +22,11 @@ buildGoModule rec {
     # some tests rely on external tools we'd rather not depend on
     rm internal/tests/integration/datasources_consul_test.go \
       internal/tests/integration/datasources_vault*_test.go
+  '';
+
+  # TestInputDir_RespectsUlimit
+  preCheck = ''
+    ulimit -n 1024
   '';
 
   ldflags = [

--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -23709,10 +23709,7 @@ with pkgs;
 
   gomp = callPackage ../applications/version-management/gomp { };
 
-  gomplate = callPackage ../development/tools/gomplate {
-    # pinned due to build failure or vendoring problems. When unpinning double check with: nix-build -A $name.go-modules --rebuild
-    buildGoModule = buildGo117Module;
-  };
+  gomplate = callPackage ../development/tools/gomplate { };
 
   gpm = callPackage ../servers/gpm {
     ncurses = null;  # Keep curses disabled for lack of value

